app/controllers/requests_controller.rb

```diff
@@ -84,20 +84,19 @@ def car_pick_up
   def car_parked
     @request = Request.find_by!('valet_pick_up_id = ? AND id = ?', Valet.find(params[:valet_id]).id, params[:request_id])
     @request.update(bay_number: valet_car_parked_params[:bay_number])
-    @request.keys_dropped!
     parking_spot = {
       latitude:   @request.parking_location.latitude,
       longitude:  @request.parking_location.longitude,
       address:    @request.parking_location.address
     }
     PrivatePub.publish_to "/user/#{@request.id}", :parking_spot => parking_spot 
+    @request.keys_dropped!
```

Comment:
Nope, but it works when i moved it down.  I think I was just updating the status too early before I validate the update actually goes through?  So far i haven't had an issue